### PR TITLE
Add mdformat plugins

### PR DESCRIPTION
**Pull request by aldoborrero**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Previous related PRs:
- https://github.com/NixOS/nixpkgs/pull/234916
- https://github.com/NixOS/nixpkgs/pull/198094 
- https://github.com/NixOS/nixpkgs/pull/198095 
- https://github.com/NixOS/nixpkgs/pull/198096 
- https://github.com/NixOS/nixpkgs/pull/198098

### CODE REVIEW

**Review by polarmutex — APPROVED** [review]

**Review by zimbatm — APPROVED** [review]

**Review by dotlambda — COMMENTED** [review]
Any reason you use `buildInputs` rather than `propagatedBuildInputs`? See https://github.com/NixOS/nixpkgs/issues/287074.
Why are the commit messages not prefixed with `python311Packages.`?
This should be fixed or the commits reverted asap.

File: pkgs/development/python-modules/mdformat-admon/default.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, mdformat
+, python3
+, pythonOlder
+}:
+
+let
+  python = python3.override {
```
Comment (dotlambda) [inline pkgs/development/python-modules/mdformat-admon/default.nix:10]: This is not okay inside of `pythonPackages`.

**Comment by dotlambda** [comment]
I'm fixing this mess in https://github.com/NixOS/nixpkgs/pull/287103.

**Comment by aldoborrero** [comment]
> Any reason you use `buildInputs` rather than `propagatedBuildInputs`? See #287074. 

At the time, I was learning how to package Python apps properly. So basically mistakes on my part.

> Why are the commit messages not prefixed with `python311Packages.`? 

I was not aware, again, at that time.

File: pkgs/development/python-modules/mdformat-admon/default.nix (hunk 1 above)
Comment (aldoborrero) [reply pkgs/development/python-modules/mdformat-admon/default.nix:10]: Yep, never again ;)

**Pull request merged**